api/src/test/java/org/openmrs/api/CohortServiceTest.java

```diff
@@ -564,8 +574,194 @@ public void getMemberships_shouldGetMembershipsAsOfADate() throws ParseException
+		cohort.addMembership(voidedMembership);
+
+		Context.getCohortService().saveCohort(cohort);
+		List<CohortMembership> unvoidedMemberships = cohort.getMemberships(false);
+		assertEquals(unvoidedMemberships.size(), 2);
+	}
+
+	/**
+	 * @verifies not return ended memberships
+	 * @see CohortService#getCohortsContainingPatient(org.openmrs.Patient)
+	 */
+	@Test
+	public void getCohortsContainingPatient_shouldNotReturnEndedMemberships() throws Exception {
+		executeDataSet(COHORT_XML);
+		
+		Cohort cohort = service.getCohort(2);
+		
+		Patient patient = new Patient(7);
+		CohortMembership membership = new CohortMembership(patient);
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		membership.setStartDate(startDate);
+		cohort.addMembership(membership);
+		assertTrue(cohort.contains(patient.getPatientId()));
+		
+		Date endDate = dateFormat.parse("2017-01-31 00:00:00");
+		membership.setEndDate(endDate);
+		
+		List<Cohort> cohortsWithPatientAdded = service.getCohortsContainingPatient(patient);
+		assertEquals(cohortsWithPatientAdded.size(), 0);
+	}
+	
+	/**
+	 * @verifies not return memberships outside a specified date
+	 * @see CohortService#getCohortsContainingPatient(org.openmrs.Patient)
+	 */
+	@Test
+	public void getCohortsContainingPatient_shouldNotReturnMembershipsOutsideSpecifiedDate() throws Exception {
+		executeDataSet(COHORT_XML);
+		
+		Cohort cohort = service.getCohort(2);
+		
+		Patient patient = new Patient(7);
+		CohortMembership membership = new CohortMembership(patient);
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		membership.setStartDate(startDate);
+		cohort.addMembership(membership);
+		assertTrue(cohort.contains(patient.getPatientId()));
+		
+		Date date = dateFormat.parse("2016-01-01 00:00:00");
+		assertEquals(service.getCohortsContainingPatient(patient, false, date).size(), 0);
+	}
+	
+	/**
+	 * @verifies return memberships on specified date
+	 * @see CohortService#getCohortsContainingPatient(org.openmrs.Patient)
+	 */
+	@Test
+	public void getCohortsContainingPatient_shouldReturnMembershipsOnSpecifiedDate() throws Exception {
+		executeDataSet(COHORT_XML);
+		
+		Cohort cohort = service.getCohort(2);
+		
+		Patient patient = new Patient(7);
+		CohortMembership membership = new CohortMembership(patient);
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		membership.setStartDate(startDate);
+		cohort.addMembership(membership);
+		assertTrue(cohort.contains(patient.getPatientId()));
+		
+		Date date = dateFormat.parse("2017-01-01 00:00:00");
+		assertEquals(service.getCohortsContainingPatient(patient, false, date).size(), 1);
+	}
+
+	@Test
+	public void union_shouldContainVoidedAndExpiredMemberships() throws Exception {
```

Comment:
These tests for union/intersection/subtract etc belong to CohortTest.java (add it if it doesn't exist) because the methods are found in Cohort.java.